python/core/auto_generated/vector/qgsvectorlayer.sip.in

```diff
@@ -2636,6 +2636,19 @@ Configuration and logic to apply automatically on any edit happening on this lay
 Returns the manager of the stored expressions for this layer.
 
 .. versionadded:: 3.10
+%End
+
+    bool mergeSelectedFeatures( const QgsAttributes &mergeAttributes, QgsGeometry unionGeometry, QString &errorMessage );
```

Comment:
This belongs in QgsVectorLayerEditUtils, not QgsVectorLayer

Reply:
Make sense thanks.
Does it make sense to just merge the selected features or would it better to explicitly pass the features to be merged? I think the second one is more flexible but maybe there are some drawbacks I can't think of...
In the second case should the feature be passed as a list of features or a list of features id?

Reply:
> explicitly pass the features to be merged

that one. A list of feature ids will be more flexible IMO.